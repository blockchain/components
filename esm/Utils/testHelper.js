import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../Tools/theme';

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(React.createElement(ThemeProvider, {
    theme: theme
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

export function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [shallow].concat(Array.prototype.slice.call(arguments)));
}
export function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [mount].concat(Array.prototype.slice.call(arguments)));
}