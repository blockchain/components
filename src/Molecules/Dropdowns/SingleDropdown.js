// @flow strict
import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { InputLabel } from '../../Atoms/Inputs/Input'
import {
  fontSize,
  prop,
  theme,
  type ThemePropType,
} from '../../Tools/interpolation'

type PropsType = {
  +capitalize?: boolean,
  +className?: string,
  +disabled?: boolean,
  +error?: string,
  +input?: {},
  +italic?: boolean,
  +items: Array<{| +text: string, +value: string | number |}>,
  +label?: string,
  +name?: string,
  +size?: string,
  +uppercase?: boolean,
  +value?: string | number,
  +weight?: 100 | 200 | 300 | 400 | 500 | 600 | 700 | 800 | 900,
  +width?: string,
}

const Wrapper = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  height: 100%;
  width: ${prop<PropsType>('width')};
  box-sizing: border-box;

  &::after {
    content: '';
    position: absolute;
    right: 1rem;
    bottom: 1.15rem;
    width: 0;
    height: 0;
    border-left: 0.4rem solid transparent;
    border-right: 0.4rem solid transparent;
    border-top: 0.4rem solid ${theme('inputColor')};
    pointer-events: none;
  }
`

const Select = styled.select`
  width: 100%;
  height: 2.7rem;
  padding: 0 2.4rem 0 1rem;
  color: ${theme('inputColor')};
  background-color: ${theme('inputBackgroundColor')};
  appearance: none;
  font-family: ${theme('fontPrimary')};
  font-weight: ${prop<PropsType>('weight')};
  font-size: ${fontSize('md')};
  font-style: ${(props: PropsType) =>
    props.italic === true ? 'italic' : 'normal'};
  border-width: 1px;
  border-color: ${(props: PropsType & ThemePropType) =>
    props.error != null
      ? theme('dangerColor')(props)
      : theme('inputBorderColor')(props)};
  border-style: solid;
  border-radius: ${theme('inputBorderRadius')};
  cursor: pointer;
  outline: none;
  text-overflow: ellipsis;
  white-space: nowrap;
  transition: box-shadow 0.25s ease-out;

  &:disabled {
    color: ${theme('inputColor')};
    border-color: ${theme('inputBorderColor')};
    background: ${theme('inputBackgroundColorDisabled')};
    cursor: not-allowed;
  }

  &:active:not(:disabled),
  &:focus:not(:disabled) {
    box-shadow: 0 0 0 1px
      ${(props: PropsType & ThemePropType) =>
        props.error != null ? 'transparent' : theme('inputActiveColor')};
  }
`

const SingleDropdown = ({
  className,
  input,
  items,
  width,
  ...rest
}: PropsType) => {
  const hasLabel = rest.label != null
  const hasError = rest.error != null

  return (
    <Wrapper className={className} width={width}>
      {(hasLabel || hasError) && (
        <InputLabel disabled={rest.disabled} error={hasError} name={rest.name}>
          {`${rest.label || ''}${
            hasLabel && hasError ? ' - ' : ''
          }${rest.error || ''}`}
        </InputLabel>
      )}
      <Select {...input} {...rest}>
        {items.map((item) => (
          <option key={item.value} value={item.value}>
            {item.text}
          </option>
        ))}
      </Select>
    </Wrapper>
  )
}

SingleDropdown.propTypes = {
  capitalize: PropTypes.bool,
  italic: PropTypes.bool,
  items: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string,
      value: PropTypes.any,
    }),
  ).isRequired,
  uppercase: PropTypes.bool,
  value: PropTypes.any,
  weight: PropTypes.oneOf([100, 200, 300, 400, 500, 600, 700, 800, 900]),
  width: PropTypes.string.isRequired,
}

SingleDropdown.defaultProps = {
  capitalize: false,
  italic: false,
  uppercase: false,
  width: '100%',
}

export default SingleDropdown
