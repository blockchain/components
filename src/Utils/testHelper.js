import React from 'react'
import { shallow, mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../Tools/theme'

function wrapWithTheme(fn, children, options) {
  const wrapper = fn(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    options,
  )

  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext(),
  })
}

export function shallowWithTheme() {
  return wrapWithTheme(shallow, ...arguments)
}

export function mountWithTheme() {
  return wrapWithTheme(mount, ...arguments)
}
