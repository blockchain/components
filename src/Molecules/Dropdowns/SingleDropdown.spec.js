import React from 'react'
import { mountWithTheme } from '../../Utils/testHelper'

import SingleDropdown from './SingleDropdown'

const items = [
  { text: 'Item1', value: 1 },
  { text: 'Item2', value: 2 },
  { text: 'Item3', value: 3 },
  { text: 'Item4', value: 4 },
  { text: 'Item5', value: 5 },
  { text: 'Item6', value: 6 },
  { text: 'Item7', value: 7 },
  { text: 'Item8', value: 8 },
  { text: 'Item9', value: 9 },
]

describe('SingleDropdown', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(
      <SingleDropdown items={items} onChange={jest.fn()} />,
    )

    expect(tree.find(SingleDropdown)).toMatchSnapshot()
  })

  it('should render correctly with capitalize', () => {
    const tree = mountWithTheme(
      <SingleDropdown capitalize items={items} onChange={jest.fn()} />,
    )

    expect(tree.find(SingleDropdown)).toMatchSnapshot()
  })

  it('should render correctly with inline', () => {
    const tree = mountWithTheme(
      <SingleDropdown inline items={items} onChange={jest.fn()} />,
    )

    expect(tree.find(SingleDropdown)).toMatchSnapshot()
  })

  it('should render correctly with italic', () => {
    const tree = mountWithTheme(
      <SingleDropdown italic items={items} onChange={jest.fn()} />,
    )

    expect(tree.find(SingleDropdown)).toMatchSnapshot()
  })

  it('should render correctly with size', () => {
    const tree = mountWithTheme(
      <SingleDropdown items={items} onChange={jest.fn()} size="32px" />,
    )

    expect(tree.find(SingleDropdown)).toMatchSnapshot()
  })

  it('should render correctly with uppercase', () => {
    const tree = mountWithTheme(
      <SingleDropdown items={items} onChange={jest.fn()} uppercase />,
    )

    expect(tree.find(SingleDropdown)).toMatchSnapshot()
  })

  it('should render correctly with weight', () => {
    const tree = mountWithTheme(
      <SingleDropdown items={items} onChange={jest.fn()} weight={500} />,
    )

    expect(tree.find(SingleDropdown)).toMatchSnapshot()
  })
})
