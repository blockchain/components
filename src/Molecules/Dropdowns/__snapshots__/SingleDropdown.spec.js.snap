// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleDropdown should render correctly 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 2.7rem;
  box-sizing: border-box;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
}

.c0::after {
  content: '';
  position: absolute;
  right: 1rem;
  top: 50%;
  margin-top: -3px;
  margin-left: 5px;
  text-align: center;
  width: 0;
  height: 0;
  border-left: 0.4rem solid transparent;
  border-right: 0.4rem solid transparent;
  border-top: 0.4rem solid #545456;
  pointer-events: none;
}

.c0:hover {
  background-color: inherit;
}

.c1 {
  position: relative;
  width: 100%;
  height: 100%;
  padding: 0 2.4rem 0 1.4rem;
  color: #545456;
  background-color: transparent;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  text-transform: none;
  font-style: normal;
  border-width: 1px;
  border-color: #CCC;
  border-style: solid;
  border-radius: 0.125rem;
  border-radius: 0.125rem;
  cursor: pointer;
  outline: none;
  text-align: left;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c1:disabled {
  cursor: not-allowed;
}

.c1:focus {
  border-color: #CCC;
}

<SingleDropdown
  capitalize={false}
  italic={false}
  items={
    Array [
      Object {
        "text": "Item1",
        "value": 1,
      },
      Object {
        "text": "Item2",
        "value": 2,
      },
      Object {
        "text": "Item3",
        "value": 3,
      },
      Object {
        "text": "Item4",
        "value": 4,
      },
      Object {
        "text": "Item5",
        "value": 5,
      },
      Object {
        "text": "Item6",
        "value": 6,
      },
      Object {
        "text": "Item7",
        "value": 7,
      },
      Object {
        "text": "Item8",
        "value": 8,
      },
      Object {
        "text": "Item9",
        "value": 9,
      },
    ]
  }
  onChange={[MockFunction]}
  size="1rem"
  uppercase={false}
  width="100%"
>
  <styled.div
    width="100%"
  >
    <div
      className="c0"
      width="100%"
    >
      <styled.select
        capitalize={false}
        italic={false}
        onChange={[MockFunction]}
        size="1rem"
        uppercase={false}
      >
        <select
          className="c1"
          onChange={[MockFunction]}
          size="1rem"
        >
          <option
            key="1"
            value={1}
          >
            Item1
          </option>
          <option
            key="2"
            value={2}
          >
            Item2
          </option>
          <option
            key="3"
            value={3}
          >
            Item3
          </option>
          <option
            key="4"
            value={4}
          >
            Item4
          </option>
          <option
            key="5"
            value={5}
          >
            Item5
          </option>
          <option
            key="6"
            value={6}
          >
            Item6
          </option>
          <option
            key="7"
            value={7}
          >
            Item7
          </option>
          <option
            key="8"
            value={8}
          >
            Item8
          </option>
          <option
            key="9"
            value={9}
          >
            Item9
          </option>
        </select>
      </styled.select>
    </div>
  </styled.div>
</SingleDropdown>
`;

exports[`SingleDropdown should render correctly with capitalize 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 2.7rem;
  box-sizing: border-box;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
}

.c0::after {
  content: '';
  position: absolute;
  right: 1rem;
  top: 50%;
  margin-top: -3px;
  margin-left: 5px;
  text-align: center;
  width: 0;
  height: 0;
  border-left: 0.4rem solid transparent;
  border-right: 0.4rem solid transparent;
  border-top: 0.4rem solid #545456;
  pointer-events: none;
}

.c0:hover {
  background-color: inherit;
}

.c1 {
  position: relative;
  width: 100%;
  height: 100%;
  padding: 0 2.4rem 0 1.4rem;
  color: #545456;
  background-color: transparent;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  text-transform: capitalize;
  font-style: normal;
  border-width: 1px;
  border-color: #CCC;
  border-style: solid;
  border-radius: 0.125rem;
  border-radius: 0.125rem;
  cursor: pointer;
  outline: none;
  text-align: left;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c1:disabled {
  cursor: not-allowed;
}

.c1:focus {
  border-color: #CCC;
}

<SingleDropdown
  capitalize={true}
  italic={false}
  items={
    Array [
      Object {
        "text": "Item1",
        "value": 1,
      },
      Object {
        "text": "Item2",
        "value": 2,
      },
      Object {
        "text": "Item3",
        "value": 3,
      },
      Object {
        "text": "Item4",
        "value": 4,
      },
      Object {
        "text": "Item5",
        "value": 5,
      },
      Object {
        "text": "Item6",
        "value": 6,
      },
      Object {
        "text": "Item7",
        "value": 7,
      },
      Object {
        "text": "Item8",
        "value": 8,
      },
      Object {
        "text": "Item9",
        "value": 9,
      },
    ]
  }
  onChange={[MockFunction]}
  size="1rem"
  uppercase={false}
  width="100%"
>
  <styled.div
    width="100%"
  >
    <div
      className="c0"
      width="100%"
    >
      <styled.select
        capitalize={true}
        italic={false}
        onChange={[MockFunction]}
        size="1rem"
        uppercase={false}
      >
        <select
          className="c1"
          onChange={[MockFunction]}
          size="1rem"
        >
          <option
            key="1"
            value={1}
          >
            Item1
          </option>
          <option
            key="2"
            value={2}
          >
            Item2
          </option>
          <option
            key="3"
            value={3}
          >
            Item3
          </option>
          <option
            key="4"
            value={4}
          >
            Item4
          </option>
          <option
            key="5"
            value={5}
          >
            Item5
          </option>
          <option
            key="6"
            value={6}
          >
            Item6
          </option>
          <option
            key="7"
            value={7}
          >
            Item7
          </option>
          <option
            key="8"
            value={8}
          >
            Item8
          </option>
          <option
            key="9"
            value={9}
          >
            Item9
          </option>
        </select>
      </styled.select>
    </div>
  </styled.div>
</SingleDropdown>
`;

exports[`SingleDropdown should render correctly with inline 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 2.7rem;
  box-sizing: border-box;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
}

.c0::after {
  content: '';
  position: absolute;
  right: 1rem;
  top: 50%;
  margin-top: -3px;
  margin-left: 5px;
  text-align: center;
  width: 0;
  height: 0;
  border-left: 0.4rem solid transparent;
  border-right: 0.4rem solid transparent;
  border-top: 0.4rem solid #545456;
  pointer-events: none;
}

.c0:hover {
  background-color: #F8F8F8;
}

.c1 {
  position: relative;
  width: 100%;
  height: 100%;
  padding: 0 2.4rem 0 1.4rem;
  color: #545456;
  background-color: transparent;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  text-transform: none;
  font-style: normal;
  border-width: 1px;
  border-color: transparent;
  border-style: solid;
  border-radius: 0.125rem;
  border-radius: 0.125rem;
  cursor: pointer;
  outline: none;
  text-align: left;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c1:disabled {
  cursor: not-allowed;
}

.c1:focus {
  border-color: #CCC;
}

<SingleDropdown
  capitalize={false}
  inline={true}
  italic={false}
  items={
    Array [
      Object {
        "text": "Item1",
        "value": 1,
      },
      Object {
        "text": "Item2",
        "value": 2,
      },
      Object {
        "text": "Item3",
        "value": 3,
      },
      Object {
        "text": "Item4",
        "value": 4,
      },
      Object {
        "text": "Item5",
        "value": 5,
      },
      Object {
        "text": "Item6",
        "value": 6,
      },
      Object {
        "text": "Item7",
        "value": 7,
      },
      Object {
        "text": "Item8",
        "value": 8,
      },
      Object {
        "text": "Item9",
        "value": 9,
      },
    ]
  }
  onChange={[MockFunction]}
  size="1rem"
  uppercase={false}
  width="100%"
>
  <styled.div
    inline={true}
    width="100%"
  >
    <div
      className="c0"
      width="100%"
    >
      <styled.select
        capitalize={false}
        inline={true}
        italic={false}
        onChange={[MockFunction]}
        size="1rem"
        uppercase={false}
      >
        <select
          className="c1"
          onChange={[MockFunction]}
          size="1rem"
        >
          <option
            key="1"
            value={1}
          >
            Item1
          </option>
          <option
            key="2"
            value={2}
          >
            Item2
          </option>
          <option
            key="3"
            value={3}
          >
            Item3
          </option>
          <option
            key="4"
            value={4}
          >
            Item4
          </option>
          <option
            key="5"
            value={5}
          >
            Item5
          </option>
          <option
            key="6"
            value={6}
          >
            Item6
          </option>
          <option
            key="7"
            value={7}
          >
            Item7
          </option>
          <option
            key="8"
            value={8}
          >
            Item8
          </option>
          <option
            key="9"
            value={9}
          >
            Item9
          </option>
        </select>
      </styled.select>
    </div>
  </styled.div>
</SingleDropdown>
`;

exports[`SingleDropdown should render correctly with italic 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 2.7rem;
  box-sizing: border-box;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
}

.c0::after {
  content: '';
  position: absolute;
  right: 1rem;
  top: 50%;
  margin-top: -3px;
  margin-left: 5px;
  text-align: center;
  width: 0;
  height: 0;
  border-left: 0.4rem solid transparent;
  border-right: 0.4rem solid transparent;
  border-top: 0.4rem solid #545456;
  pointer-events: none;
}

.c0:hover {
  background-color: inherit;
}

.c1 {
  position: relative;
  width: 100%;
  height: 100%;
  padding: 0 2.4rem 0 1.4rem;
  color: #545456;
  background-color: transparent;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  text-transform: none;
  font-style: italic;
  border-width: 1px;
  border-color: #CCC;
  border-style: solid;
  border-radius: 0.125rem;
  border-radius: 0.125rem;
  cursor: pointer;
  outline: none;
  text-align: left;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c1:disabled {
  cursor: not-allowed;
}

.c1:focus {
  border-color: #CCC;
}

<SingleDropdown
  capitalize={false}
  italic={true}
  items={
    Array [
      Object {
        "text": "Item1",
        "value": 1,
      },
      Object {
        "text": "Item2",
        "value": 2,
      },
      Object {
        "text": "Item3",
        "value": 3,
      },
      Object {
        "text": "Item4",
        "value": 4,
      },
      Object {
        "text": "Item5",
        "value": 5,
      },
      Object {
        "text": "Item6",
        "value": 6,
      },
      Object {
        "text": "Item7",
        "value": 7,
      },
      Object {
        "text": "Item8",
        "value": 8,
      },
      Object {
        "text": "Item9",
        "value": 9,
      },
    ]
  }
  onChange={[MockFunction]}
  size="1rem"
  uppercase={false}
  width="100%"
>
  <styled.div
    width="100%"
  >
    <div
      className="c0"
      width="100%"
    >
      <styled.select
        capitalize={false}
        italic={true}
        onChange={[MockFunction]}
        size="1rem"
        uppercase={false}
      >
        <select
          className="c1"
          onChange={[MockFunction]}
          size="1rem"
        >
          <option
            key="1"
            value={1}
          >
            Item1
          </option>
          <option
            key="2"
            value={2}
          >
            Item2
          </option>
          <option
            key="3"
            value={3}
          >
            Item3
          </option>
          <option
            key="4"
            value={4}
          >
            Item4
          </option>
          <option
            key="5"
            value={5}
          >
            Item5
          </option>
          <option
            key="6"
            value={6}
          >
            Item6
          </option>
          <option
            key="7"
            value={7}
          >
            Item7
          </option>
          <option
            key="8"
            value={8}
          >
            Item8
          </option>
          <option
            key="9"
            value={9}
          >
            Item9
          </option>
        </select>
      </styled.select>
    </div>
  </styled.div>
</SingleDropdown>
`;

exports[`SingleDropdown should render correctly with size 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 2.7rem;
  box-sizing: border-box;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
}

.c0::after {
  content: '';
  position: absolute;
  right: 1rem;
  top: 50%;
  margin-top: -3px;
  margin-left: 5px;
  text-align: center;
  width: 0;
  height: 0;
  border-left: 0.4rem solid transparent;
  border-right: 0.4rem solid transparent;
  border-top: 0.4rem solid #545456;
  pointer-events: none;
}

.c0:hover {
  background-color: inherit;
}

.c1 {
  position: relative;
  width: 100%;
  height: 100%;
  padding: 0 2.4rem 0 1.4rem;
  color: #545456;
  background-color: transparent;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 32px;
  text-transform: none;
  font-style: normal;
  border-width: 1px;
  border-color: #CCC;
  border-style: solid;
  border-radius: 0.125rem;
  border-radius: 0.125rem;
  cursor: pointer;
  outline: none;
  text-align: left;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c1:disabled {
  cursor: not-allowed;
}

.c1:focus {
  border-color: #CCC;
}

<SingleDropdown
  capitalize={false}
  italic={false}
  items={
    Array [
      Object {
        "text": "Item1",
        "value": 1,
      },
      Object {
        "text": "Item2",
        "value": 2,
      },
      Object {
        "text": "Item3",
        "value": 3,
      },
      Object {
        "text": "Item4",
        "value": 4,
      },
      Object {
        "text": "Item5",
        "value": 5,
      },
      Object {
        "text": "Item6",
        "value": 6,
      },
      Object {
        "text": "Item7",
        "value": 7,
      },
      Object {
        "text": "Item8",
        "value": 8,
      },
      Object {
        "text": "Item9",
        "value": 9,
      },
    ]
  }
  onChange={[MockFunction]}
  size="32px"
  uppercase={false}
  width="100%"
>
  <styled.div
    width="100%"
  >
    <div
      className="c0"
      width="100%"
    >
      <styled.select
        capitalize={false}
        italic={false}
        onChange={[MockFunction]}
        size="32px"
        uppercase={false}
      >
        <select
          className="c1"
          onChange={[MockFunction]}
          size="32px"
        >
          <option
            key="1"
            value={1}
          >
            Item1
          </option>
          <option
            key="2"
            value={2}
          >
            Item2
          </option>
          <option
            key="3"
            value={3}
          >
            Item3
          </option>
          <option
            key="4"
            value={4}
          >
            Item4
          </option>
          <option
            key="5"
            value={5}
          >
            Item5
          </option>
          <option
            key="6"
            value={6}
          >
            Item6
          </option>
          <option
            key="7"
            value={7}
          >
            Item7
          </option>
          <option
            key="8"
            value={8}
          >
            Item8
          </option>
          <option
            key="9"
            value={9}
          >
            Item9
          </option>
        </select>
      </styled.select>
    </div>
  </styled.div>
</SingleDropdown>
`;

exports[`SingleDropdown should render correctly with uppercase 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 2.7rem;
  box-sizing: border-box;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
}

.c0::after {
  content: '';
  position: absolute;
  right: 1rem;
  top: 50%;
  margin-top: -3px;
  margin-left: 5px;
  text-align: center;
  width: 0;
  height: 0;
  border-left: 0.4rem solid transparent;
  border-right: 0.4rem solid transparent;
  border-top: 0.4rem solid #545456;
  pointer-events: none;
}

.c0:hover {
  background-color: inherit;
}

.c1 {
  position: relative;
  width: 100%;
  height: 100%;
  padding: 0 2.4rem 0 1.4rem;
  color: #545456;
  background-color: transparent;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  text-transform: uppercase;
  font-style: normal;
  border-width: 1px;
  border-color: #CCC;
  border-style: solid;
  border-radius: 0.125rem;
  border-radius: 0.125rem;
  cursor: pointer;
  outline: none;
  text-align: left;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c1:disabled {
  cursor: not-allowed;
}

.c1:focus {
  border-color: #CCC;
}

<SingleDropdown
  capitalize={false}
  italic={false}
  items={
    Array [
      Object {
        "text": "Item1",
        "value": 1,
      },
      Object {
        "text": "Item2",
        "value": 2,
      },
      Object {
        "text": "Item3",
        "value": 3,
      },
      Object {
        "text": "Item4",
        "value": 4,
      },
      Object {
        "text": "Item5",
        "value": 5,
      },
      Object {
        "text": "Item6",
        "value": 6,
      },
      Object {
        "text": "Item7",
        "value": 7,
      },
      Object {
        "text": "Item8",
        "value": 8,
      },
      Object {
        "text": "Item9",
        "value": 9,
      },
    ]
  }
  onChange={[MockFunction]}
  size="1rem"
  uppercase={true}
  width="100%"
>
  <styled.div
    width="100%"
  >
    <div
      className="c0"
      width="100%"
    >
      <styled.select
        capitalize={false}
        italic={false}
        onChange={[MockFunction]}
        size="1rem"
        uppercase={true}
      >
        <select
          className="c1"
          onChange={[MockFunction]}
          size="1rem"
        >
          <option
            key="1"
            value={1}
          >
            Item1
          </option>
          <option
            key="2"
            value={2}
          >
            Item2
          </option>
          <option
            key="3"
            value={3}
          >
            Item3
          </option>
          <option
            key="4"
            value={4}
          >
            Item4
          </option>
          <option
            key="5"
            value={5}
          >
            Item5
          </option>
          <option
            key="6"
            value={6}
          >
            Item6
          </option>
          <option
            key="7"
            value={7}
          >
            Item7
          </option>
          <option
            key="8"
            value={8}
          >
            Item8
          </option>
          <option
            key="9"
            value={9}
          >
            Item9
          </option>
        </select>
      </styled.select>
    </div>
  </styled.div>
</SingleDropdown>
`;

exports[`SingleDropdown should render correctly with weight 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 2.7rem;
  box-sizing: border-box;
  background-color: #FFF;
  background-image: none;
  outline-width: 0;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
}

.c0::after {
  content: '';
  position: absolute;
  right: 1rem;
  top: 50%;
  margin-top: -3px;
  margin-left: 5px;
  text-align: center;
  width: 0;
  height: 0;
  border-left: 0.4rem solid transparent;
  border-right: 0.4rem solid transparent;
  border-top: 0.4rem solid #545456;
  pointer-events: none;
}

.c0:hover {
  background-color: inherit;
}

.c1 {
  position: relative;
  width: 100%;
  height: 100%;
  padding: 0 2.4rem 0 1.4rem;
  color: #545456;
  background-color: transparent;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-size: 1rem;
  font-weight: 500;
  text-transform: none;
  font-style: normal;
  border-width: 1px;
  border-color: #CCC;
  border-style: solid;
  border-radius: 0.125rem;
  border-radius: 0.125rem;
  cursor: pointer;
  outline: none;
  text-align: left;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.c1:disabled {
  cursor: not-allowed;
}

.c1:focus {
  border-color: #CCC;
}

<SingleDropdown
  capitalize={false}
  italic={false}
  items={
    Array [
      Object {
        "text": "Item1",
        "value": 1,
      },
      Object {
        "text": "Item2",
        "value": 2,
      },
      Object {
        "text": "Item3",
        "value": 3,
      },
      Object {
        "text": "Item4",
        "value": 4,
      },
      Object {
        "text": "Item5",
        "value": 5,
      },
      Object {
        "text": "Item6",
        "value": 6,
      },
      Object {
        "text": "Item7",
        "value": 7,
      },
      Object {
        "text": "Item8",
        "value": 8,
      },
      Object {
        "text": "Item9",
        "value": 9,
      },
    ]
  }
  onChange={[MockFunction]}
  size="1rem"
  uppercase={false}
  weight={500}
  width="100%"
>
  <styled.div
    width="100%"
  >
    <div
      className="c0"
      width="100%"
    >
      <styled.select
        capitalize={false}
        italic={false}
        onChange={[MockFunction]}
        size="1rem"
        uppercase={false}
        weight={500}
      >
        <select
          className="c1"
          onChange={[MockFunction]}
          size="1rem"
        >
          <option
            key="1"
            value={1}
          >
            Item1
          </option>
          <option
            key="2"
            value={2}
          >
            Item2
          </option>
          <option
            key="3"
            value={3}
          >
            Item3
          </option>
          <option
            key="4"
            value={4}
          >
            Item4
          </option>
          <option
            key="5"
            value={5}
          >
            Item5
          </option>
          <option
            key="6"
            value={6}
          >
            Item6
          </option>
          <option
            key="7"
            value={7}
          >
            Item7
          </option>
          <option
            key="8"
            value={8}
          >
            Item8
          </option>
          <option
            key="9"
            value={9}
          >
            Item9
          </option>
        </select>
      </styled.select>
    </div>
  </styled.div>
</SingleDropdown>
`;
